app/components/primer/alpha/action_bar_element.ts

```diff
@@ -86,11 +86,18 @@ class ActionBarElement extends HTMLElement {
       bindKeys: FocusKeys.ArrowHorizontal | FocusKeys.HomeAndEnd,
       focusOutBehavior: 'wrap',
       focusableElementFilter: element => {
-        return !element.closest('.ActionBar-item[hidden]') && !element.closest('li.ActionListItem')
+        return this.#isVisible(element)
       }
     })
   }
 
+  #isVisible(element: HTMLElement): boolean {
+    // Safari doesn't support `checkVisibility` yet.
+    if (typeof element.checkVisibility === 'function') return element.checkVisibility()
```

Comment:
@jonrohan this should be easy enough to polyfill (https://drafts.csswg.org/cssom-view-1/#dom-element-checkvisibility):

```js
function checkVisibility(el, { checkOpacity = false, checkVisibilityCSS = false } = {}) {
  const rect = el.getBoundingClientRect();
  if (!rect.width || !rect.height) return false;
  let node = el;
  while (node) {
    const computed = getComputedStyle(node);
    if (computed['content-visibility'] === 'hidden') return false;
    if (checkOpacity && computed['opacity'] === '0') return false;
    node = node.parentElement;
  }
  if (checkVisibilityCSS) return getComputedStyle(el)['visibility'] !== 'hidden'
  return true;
}
```